autoload/ddc/map.vim

```diff
@@ -78,6 +78,19 @@ function ddc#map#insert_item(number, cancel_key) abort
   " Call CompleteDone later.
   autocmd ddc TextChangedI * ++once doautocmd <nomodeline> CompleteDone
 
+  " Avoid side effects of ddc-cmdline
+  if mode() ==# 'c'
+    const cmdline = getcmdline()
+    const prev_input = g:ddc#_complete_pos == 0
+          \ ? ''
+          \ : cmdline[: g:ddc#_complete_pos - 1]
+    const next_input = cmdline[getcmdpos() - 1 :]
```

Comment:
`s:col()` is better.

Reply:
Why use `s:col()` with complex conditional branching when it's already known that the mode is `c`?